src/foam/nanos/ruler/RuleEngine.java

```diff
@@ -230,14 +229,15 @@ private boolean checkPermission(Rule rule, FObject obj) {
     return true;
   }
 
-  private FObject reloadObject(FObject obj) {
-    var reloaded = getDelegate().find_(userX_, obj);
-    // For rules with operation=REMOVE, the object was removed from the DAO thus
+  private FObject reloadObject(FObject obj, Operation operation) {
+    // For REMOVE operation, the object was removed from the DAO thus
     // returning the original object.
-    if ( reloaded == null ) {
+    if ( operation == Operation.REMOVE) {
```

Comment:
This is confusing. From reading the code above, I thought you were reloading the object twice. If the method is called reloadObject(), I think it should reload the object. I would remove this operation and just put it above where it will be clearer.

Reply:
Updated.